src/webapp/components/table/performance-overview-table/PerformanceOverviewTable.tsx

```diff
@@ -0,0 +1,277 @@
+import React, { useEffect, useMemo, useState } from "react";
+import _ from "lodash";
+import {
+    Table,
+    TableBody,
+    TableCell,
+    TableHead,
+    TableRow,
+    TableContainer,
+} from "@material-ui/core";
+import styled from "styled-components";
+import { SearchInput } from "../../search-input/SearchInput";
+import { Selector } from "../../selector/Selector";
+import { Maybe } from "../../../../utils/ts-utils";
+import i18n from "../../../../utils/i18n";
+
+export type TableColumn = {
+    value: string;
+    label: string;
+    dark?: boolean;
+};
+
+interface PerformanceOverviewTableProps {
+    rows: {
+        [key: TableColumn["value"]]: string;
+    }[];
+}
+
+export const PerformanceOverviewTable: React.FC<PerformanceOverviewTableProps> = React.memo(
+    ({ rows }) => {
+        const [searchTerm, setSearchTerm] = useState<string>("");
+        const [filterValue, setFilterValue] = useState("");
+        const [filteredRows, setFilteredRows] = useState(rows);
+
+        const columns: TableColumn[] = [
+            { label: "Event", value: "event" },
+            { label: "Location", value: "location" },
+            { label: "Cases", value: "cases" },
+            { label: "Deaths", value: "deaths" },
+            { label: "Duration", value: "duration" },
+            { label: "Manager", value: "manager" },
+            { label: "Detect 7d", dark: true, value: "detect7d" },
+            { label: "Notify 1d", dark: true, value: "notify1d" },
+            { label: "ERA1", value: "era1" },
+            { label: "ERA2", value: "era2" },
+            { label: "ERA3", value: "era3" },
+            { label: "ERA4", value: "era4" },
+            { label: "ERA5", value: "era5" },
+            { label: "ERA6", value: "era6" },
+            { label: "ERA7", value: "era7" },
+            { label: "ERI", value: "eri" },
+            { label: "Respond 7d", dark: true, value: "respond7d" },
+        ];
+
+        const editRiskAssessmentColumns = [
+            "era1",
+            "era2",
+            "era3",
+            "era4",
+            "era5",
+            "era6",
+            "era7",
+            "eri",
+        ];
+        const columnRules: { [key: string]: number } = {
+            detect7d: 7,
+            notify1d: 1,
+            respond7d: 7,
+        };
+
+        const calculateColumns = [...editRiskAssessmentColumns, ..._.keys(columnRules)];
+
+        useEffect(() => {
+            if (searchTerm === "") {
+                setFilteredRows(rows);
+            } else {
+                const filtered = _.filter(rows, row => {
+                    return _.some(row, cell => {
+                        return _.includes(_.toLower(cell), _.toLower(searchTerm));
+                    });
+                });
+                setFilteredRows(filtered);
+            }
+        }, [searchTerm, rows]);
+
+        const getCellColor = (cellValue: Maybe<string>, column: TableColumn["value"]) => {
```

Comment:
maybe we can a new type of cell called ColoredCell and pass the color as a parameter. Also add a custom hook which contains all these calculate functions.

I am a little confused about whether this calculation should be in domain layer - because it is a business logic. However, it is true that it is a "view" logic. I will refer this to @anagperal as she is the expert in this.

Reply:
Thank Sneha but I'm not an expert :joy: , it's @xurxodev  ! But I usually ask him some clean architecture questions. I think you're right, maybe this calculations should not be in the component but I think it's more a question to ask to @xurxodev

Reply:
I agree with [this](https://github.com/EyeSeeTea/zebra-dev/pull/4/files#r1660403496) and maybe we should have a hook for all the component and there we can have all the calculations and filters so we pass to the component just the config of the table (columns, rows...)? And maybe in that config we can have in the data of the cell a value with "alert", "warning"  "ok" (or better names according to the meaning of the colors) and depending on these values the cell has one color or the default one?

Reply:
@9sneha-n what do you mean by type? A new component?

Should I just wrap `getCellColor` in a hook or create a custom hook that return rows updated with colors